### DISCUSSION TITLE
Height of row base on content - lable

### DISCUSSION THREAD

**Discussion by DoHoangYen** [nick-keller/react-datasheet-grid#199] [Q&A] [answered]
I want to display multiple line text in a cell, can I change height of row base content?

**Comment by nick-keller** [comment, marked as answer]
Hello @DoHoangYen , 
Rows can have different sizes using the [rowHeight](https://react-datasheet-grid.netlify.app/docs/api-reference/props#rowheight) property and passing a function `((opt: { rowData: T; rowIndex: number }) => number`

It is up to you to measure some how the ideal row height.

**Reply by imkitukov** [reply]
> Hello @DoHoangYen , All rows must be the same size using the [rozHeight](https://react-datasheet-grid.netlify.app/docs/api-reference/props#rowheight) property. You can change this dynamically if needed.
> 
> It is up to you to measure some how the ideal row height and pass it as a prop, but you will not be able to have different heights for different rows.


Hey @nick-keller, is there something new around supporting rows with different heights?
If no, is it possible to have some 'empty rows'  each with custom height, I have a case when sometimes I need to have rows like this after regular data rows, like a filler row to cover a gap.

![image](https://user-images.githubusercontent.com/44986252/228828734-745ea214-67e0-4bec-a0ee-28d5578901b0.png)

**Reply by nick-keller** [reply]
Yes, I updated my response and added one to your question

**Reply by mosha101** [reply]
When I specify a function to the rowHeight prop for DataSheetGrid or DynamicDataSheetGrid

I get an error "Cannot read properties of undefined (reading 'top')" referring to calculatedHeights.current while calculating totalSize. 

Is there anything else that I need to be doing when specifying a function instead of a number for rowHeight?

```
// node_modules/react-datasheet-grid/dist/hooks/useRowHeights.js
totalSize: (maxHeight) => {
  if (typeof rowHeight === "number") {
    return value.length * rowHeight;
  }
  const index = getRowIndex(maxHeight);
  return calculatedHeights.current[index].top + calculatedHeights.current[index].height;
}
```
